rollup/fees/rollup_fee.go

```diff
@@ -126,25 +143,46 @@ func rlpEncode(tx *types.Transaction) ([]byte, error) {
 	return raw.Bytes(), nil
 }
 
-func readGPOStorageSlots(addr common.Address, state StateDB) (*big.Int, *big.Int, *big.Int) {
-	l1BaseFee := state.GetState(addr, rcfg.L1BaseFeeSlot)
-	overhead := state.GetState(addr, rcfg.OverheadSlot)
-	scalar := state.GetState(addr, rcfg.ScalarSlot)
-	return l1BaseFee.Big(), overhead.Big(), scalar.Big()
+func readGPOStorageSlots(addr common.Address, state StateDB) gpoState {
+	var gpoState gpoState
+	gpoState.l1BaseFee = state.GetState(addr, rcfg.L1BaseFeeSlot).Big()
+	gpoState.overhead = state.GetState(addr, rcfg.OverheadSlot).Big()
+	gpoState.scalar = state.GetState(addr, rcfg.ScalarSlot).Big()
+	gpoState.l1BlobBaseFee = state.GetState(addr, rcfg.L1BlobBaseFeeSlot).Big()
+	gpoState.commitScalar = state.GetState(addr, rcfg.CommitScalarSlot).Big()
+	gpoState.blobScalar = state.GetState(addr, rcfg.BlobScalarSlot).Big()
+	return gpoState
 }
 
 // calculateEncodedL1DataFee computes the L1 fee for an RLP-encoded tx
-func calculateEncodedL1DataFee(data []byte, overhead, l1GasPrice *big.Int, scalar *big.Int) *big.Int {
-	l1GasUsed := CalculateL1GasUsed(data, overhead)
-	l1DataFee := new(big.Int).Mul(l1GasUsed, l1GasPrice)
+func calculateEncodedL1DataFee(data []byte, overhead, l1BaseFee *big.Int, scalar *big.Int) *big.Int {
+	l1GasUsed := calculateL1GasUsed(data, overhead)
+	l1DataFee := new(big.Int).Mul(l1GasUsed, l1BaseFee)
 	return mulAndScale(l1DataFee, scalar, rcfg.Precision)
 }
 
-// CalculateL1GasUsed computes the L1 gas used based on the calldata and
+// calculateEncodedL1DataFeeCurie computes the L1 fee for an RLP-encoded tx, post Curie
+func calculateEncodedL1DataFeeCurie(data []byte, l1BaseFee *big.Int, l1BlobBaseFee *big.Int, commitScalar *big.Int, blobScalar *big.Int) *big.Int {
+	// calldata component
+	calldataGas := new(big.Int).Mul(commitScalar, l1BaseFee)
```

Comment:
My thinking was that this covers all fees related to the "calldata part" of the transaction. `commitGas` sounds like it would cover all of calldata + execution + blobs.

Reply:
Updated the comment.